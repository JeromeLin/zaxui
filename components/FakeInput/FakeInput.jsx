import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

class FakeInput extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: props.value,
      isFocus: false,
    };
    this.fakeFocus = this.fakeFocus.bind(this);
    this.fakeBlur = this.fakeBlur.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    this.setState({
      value: nextProps.value,
    });
  }

  fakeFocus() {
    const { cbFocus } = this.props;
    const referInput = this.refs['referInput'];
    referInput.focus();
    this.setState({
      isFocus: true,
    });
    typeof cbFocus === 'function' && cbFocus();
  }

  fakeBlur(e) {
    const { cbBlur } = this.props;
    this.setState({
      isFocus: false,
    });
    typeof cbBlur === 'function' && cbBlur();
  }

  render() {
    const { prefixCls, className, wrapStyle, inputStyle, placeholder } = this.props;
    const { value, isFocus } = this.state;
    
    return (
      <article className={ classnames(`${prefixCls}`, className) }
               style={ wrapStyle }
               onClick={ this.fakeFocus }>
        <div className={ classnames('fake-input', {'editing': isFocus, 'placeholder': value === '' && !isFocus})}
             placeholder={ placeholder }
             style={ inputStyle }>{ value }</div>
        <div ref="referInput" tabIndex={-1}
             className="refer-input"
             onBlur={ this.fakeBlur }/>
      </article>
    );
  }
}

FakeInput.propTypes = {
  prefixCls: PropTypes.string,
  className: PropTypes.string,
  wrapStyle: PropTypes.object,
  inputStyle: PropTypes.object,
  value: PropTypes.string,
  placeholder: PropTypes.string,
  cbFocus: PropTypes.func,
  cbBlur: PropTypes.func,
};

FakeInput.defaultProps = {
  prefixCls: 'za-fakeinput',
  wrapStyle: {},
  inputStyle: {width: '100%', minWidth: '150px'},
  placeholder: '',
  value: '',
};

export default FakeInput;
