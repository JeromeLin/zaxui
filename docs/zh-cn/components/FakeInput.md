# 模拟输入框 FakeInput

[demo页面](https://zhongantecheng.github.io/zarm/#/numkeyboard)

### 引入

```js
import { FakeInput } from 'zarm';
```

### 代码演示

#### 基本用法

```jsx
<Input placeholder="请输入手机号" value="13512341234" />
```

### API

| 属性 | 类型 | 默认值 | 可选值／参数 | 说明 |
| :--- | :--- | :--- | :--- | :--- |
| prefixCls | string | za-fakeinput | | 类名前缀 |
| className | string | | | 追加类名 |
| wrapStyle | object | {} | | 包裹容器的自定义设置 |
| inputStyle | object | {} | | 伪input的自定义设置 |
| value | string |  | | 值 |
| placeholder | string | '' | | placeholder |
| cbFocus | func | 无 | | 获取焦点的回调 |
| cbBlur | func | 无 | | 失去焦点的回调 |
