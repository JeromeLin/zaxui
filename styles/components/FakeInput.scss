$prefixCls: $prefixCls-fakeinput;

.#{$prefixCls} {
  display: inline-block;
  max-width: 50%;
  height: r(30);
  line-height: r(30);
  vertical-align: middle;

  .fake-input {
    height: r(30);
    white-space: nowrap;
    overflow-x: scroll;
    text-align: left;
    &::-webkit-scrollbar {
      width: 0;
      height: 0;
    }
    &.placeholder {
      &:before {
        content: attr(placeholder);
        display: block;
        font-size: r(14);
        color: #ccc;
      }
    }
    &.editing {
      &:after{
        content: '|';
        display: inline-block;
        color: #000;
        animation: flashing .9s ease-in-out infinite;
      }
    }
  }
  .refer-input {
    height: 0px;
    margin: 0px;
    width: 0px;
    padding: 0;
    outline: none;
    border-width: 0;
  }
}

@keyframes flashing {
  0 {
    opacity: 1;
  }
  50% {
    opacity: 1;
  }
  51% {
    opacity: 0;
  }
  100% {
    opacity: 0;
  }
}
