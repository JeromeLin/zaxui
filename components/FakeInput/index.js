import FakeInput from './FakeInput';
export default FakeInput;