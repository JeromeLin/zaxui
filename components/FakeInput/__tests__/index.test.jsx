import React from 'react';
import { render, shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import FakeInput from '../index';

describe('FakeInput', () => {
  const props = {
    placeholder: '请输入手机号',
  };

  it('render correctly', () => {
    const wrapper = render(<FakeInput {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('FakeInput focus', () => {
    const cbFocus = jest.fn();
    const wrapper = mount(<FakeInput cbFocus={cbFocus}/>);
    wrapper.simulate('click');
    expect(cbFocus).toBeCalled();
    wrapper.unmount();
  });
});