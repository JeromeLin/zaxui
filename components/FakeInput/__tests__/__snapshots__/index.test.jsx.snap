// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FakeInput renders correctly 1`] = `
<article
  class="za-fakeinput"
>
  <div
    class="fake-input placeholder"
    placeholder="请输入手机号"
    style="width: 100%; min-width: 150px;"
  >
  </div>
  <div
    tabindex="-1"
    class="refer-input"
  >
  </div>
</article>
`;
